Update example service-account.yml for container group in documentation

mcen1 opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
We recently stood up a container group and the pod-manager role seems insufficient as was written in the original code. This one seemed to give us enough to make AWX launch a pod in a kubernetes cluster, attach to the pod, and follow the logs. Also, updated the spec from v1beta to v1.

<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - Docs

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
AWX 21.8.0
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->
Just updating documentation based on what I've experienced on our AWX instance in kubernetes when creating a service account in a different cluster.

<!--- Paste verbatim command output below, e.g. before and after your change -->
```
n/a
```

mcen1 committed:
Update service-account.yml

We recently stood up a container group and the pod-manager role seems insufficient as was written in the original code. This one seemed to give us enough to make AWX launch a pod in a kubernetes cluster, attach to the pod, and follow the logs.

TheRealHaoLiu commented:
```
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: containergroup-service-account
  namespace: containergroup-namespace
---
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: role-containergroup-service-account
  namespace: containergroup-namespace
rules:
- apiGroups: [""]
  resources: ["pods"]
  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
- apiGroups: [""]
  resources: ["pods/log"]
  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
- apiGroups: [""]
  resources: ["pods/attach"]
  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
---
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: role-containergroup-service-account-binding
  namespace: containergroup-namespace
subjects:
- kind: ServiceAccount
  name: containergroup-service-account
  namespace: containergroup-namespace
roleRef:
  kind: Role
  name: role-containergroup-service-account
  apiGroup: rbac.authorization.k8s.io
```

from official documentation https://docs.ansible.com/automation-controller/latest/html/administration/containers_instance_groups.html#create-a-container-group

TheRealHaoLiu commented:
would u mind updating your PR to contain the same information?

TheRealHaoLiu commented:
@mcen1 also, welcome to the AWX community~

come hang out with us on the AWX Matrix chatroom https://matrix.to/#/#awx:ansible.com

we do have a request to make a video for AWX YouTube channel about how to setup container group in AWX 
https://github.com/ansible/awx/issues/13242#issuecomment-1329765409

If you are interested in doing that 😉 

also I been working on this demo https://github.com/TheRealHaoLiu/awx-container-group-demo to use for the video

contribution welcome <3

mcen1 committed:
Updating per documentation

https://docs.ansible.com/automation-controller/latest/html/administration/containers_instance_groups.html#create-a-container-group

mcen1 commented:
> would u mind updating your PR to contain the same information?

Sure thing, done.

TheRealHaoLiu commented:
Hi @mcen1 
```
linters: commands[1]> flake8 awx awxkit awx_collection
linters: commands[2]> yamllint -s .
./docs/container_groups/service-account.yml
  25:1      error    wrong indentation: expected 2 but found 0  (indentation)
  41:1      error    wrong indentation: expected 2 but found 0  (indentation)
  48:1      error    trailing spaces  (trailing-spaces)
```

TheRealHaoLiu commented on docs/container_groups/service-account.yml:
```diff
@@ -13,30 +13,36 @@
-  name: pod-manager
+  name: role-containergroup-service-account
+  apiGroup: rbac.authorization.k8s.io
+  
```
```suggestion

```

TheRealHaoLiu replied:
fix yamlint error 
`  48:1      error    trailing spaces  (trailing-spaces)`

TheRealHaoLiu commented on docs/container_groups/service-account.yml:
```diff
@@ -13,30 +13,36 @@
 subjects:
-  - kind: ServiceAccount
-    name: awx
+- kind: ServiceAccount
```
```suggestion
  - kind: ServiceAccount
```

TheRealHaoLiu commented on docs/container_groups/service-account.yml:
```diff
@@ -13,30 +13,36 @@
 subjects:
-  - kind: ServiceAccount
-    name: awx
+- kind: ServiceAccount
+  name: containergroup-service-account
+  namespace: containergroup-namespace
```
```suggestion
  - kind: ServiceAccount
    name: containergroup-service-account
    namespace: containergroup-namespace
```

TheRealHaoLiu replied:
fix yamlint 
`  41:1      error    wrong indentation: expected 2 but found 0  (indentation)`

TheRealHaoLiu commented on docs/container_groups/service-account.yml:
```diff
@@ -13,30 +13,36 @@
-    resources: ["pods/exec"]
-    verbs: ["create"]
-
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+- apiGroups: [""]
+  resources: ["pods/log"]
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
+- apiGroups: [""]
+  resources: ["pods/attach"]
+  verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
```
```suggestion
  - apiGroups: [""]
    resources: ["pods"]
    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
  - apiGroups: [""]
    resources: ["pods/log"]
    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
  - apiGroups: [""]
    resources: ["pods/attach"]
    verbs: ["get", "list", "watch", "create", "update", "patch", "delete"]
```

TheRealHaoLiu replied:
fix yamllint 
`  25:1      error    wrong indentation: expected 2 but found 0  (indentation)`

mcen1 committed:
Update docs/container_groups/service-account.yml

Co-authored-by: Hao Liu <44379968+TheRealHaoLiu@users.noreply.github.com>

mcen1 committed:
Update docs/container_groups/service-account.yml

Co-authored-by: Hao Liu <44379968+TheRealHaoLiu@users.noreply.github.com>

mcen1 committed:
Update docs/container_groups/service-account.yml

Co-authored-by: Hao Liu <44379968+TheRealHaoLiu@users.noreply.github.com>

fosterseth commented:
@TheRealHaoLiu the linters were addressed recently, so just needs to pass CI and final review from dev, then we can squash and merge

mcen1 commented:
Hello all. Is there anything needed on my side for this merge? Thanks!

mcen1 commented:
Hi. Anything else needed for this to be merged?

masbahnana committed:
Merge branch 'devel' into patch-2
